src/backend/api/meals.js

```diff
@@ -6,10 +6,62 @@ const date = new Date();
 const formattedCurrentDate = date.toISOString().slice(0, 19).replace('T', ' ');
 
 router.get("/meals", async (request, response) => {
+  const allowedKeys = [
+    'meal_when',
+    'max_reservations',
+    'price',
+  ];
+
   try {
-    const meals = await knex("meals").select("*");
+    const queryParams = request.query;
+    let queryBuilder = knex("meals");
+
+    if (queryParams.maxPrice) {
+      queryBuilder.where('price', '<', queryParams.maxPrice);
+    }
+
+    if (queryParams.limit) {
+      queryBuilder.limit(queryParams.limit);
+    }
+
+    if (queryParams.title) {
+      queryBuilder.where('title', 'LIKE', `%${queryParams.title}%`);
+    }
+
+    if (queryParams.dateAfter) {
+      queryBuilder.where('meal_when', '>', queryParams.dateAfter);
+    }
+
+    if (queryParams.dateBefore) {
+      queryBuilder.where('meal_when', '<', queryParams.dateBefore);
+    }
+
+    if (queryParams.sortDir && queryParams.sortKey) {
+      if (allowedKeys.includes(queryParams.sortKey)) {
+        queryBuilder.orderBy(queryParams.sortKey, queryParams.sortDir.toUpperCase());
+      }
+    }
+
+    if (queryParams.sortKey && !queryParams.sortDir) {
+      if (allowedKeys.includes(queryParams.sortKey)) {
+        queryBuilder.orderBy(queryParams.sortKey, 'ASC');
+      }
+    }
+
+    if (queryParams.availableReservations !== undefined) {
```

Comment:
I would suggest removing the `!== undefined` part, so it also checks against null.

In JavaScript, it is usually a good idea to check for both undefined and null (and sometimes also empty string), because of the nature of client-server and user-computer communication.

You can make use of the Falsy/Truthy concept i.e. just treat the variable as a boolean: 
https://developer.mozilla.org/en-US/docs/Glossary/Falsy